services/system/Transact/src/RTransact.cpp

```diff
@@ -184,122 +184,244 @@ void RTransact::onTrx(const Checksum256& id, psio::view<const TransactionTrace>
    check(getSender() == AccountNumber{}, "Wrong sender");
    printf("trace size: %zu\n", find_view_span(trace).size());
 
-   TransactionTraceRef pruned = PruneTrace{true}(trace);
+   auto    clients = Subjective{}.open<TraceClientTable>();
+   uint8_t waitFor = WaitFor::final_flag;
+   PSIBASE_SUBJECTIVE_TX
+   {
+      if (auto row = clients.get(id))
+      {
+         waitFor = row->waitFor;
+      }
+   }
+
+   if (waitFor == WaitFor::final_flag)
+   {
+      auto currentBlock = to<Transact>().currentBlock();
+      auto blockTxs     = Subjective{}.open<BlockTxsTable>();
+      PSIBASE_SUBJECTIVE_TX
+      {
+         blockTxs.put(BlockTxRecord{
```

Comment:
This can send the wrong trace in some fork switching cases. For example,
1) apply to fork A
2) fail to apply to fork B
3) switch back to fork A and make it irreversible

Reply:
What is the right way to handle this? We don't have a subjective db that follows forks, right?